code/API_definitions/qod-api.yaml

```diff
@@ -369,9 +349,14 @@ components:
     ErrorInfo:
       type: object
       required:
+        - status
         - code
         - message
       properties:
+        status:
+          type: string
+          pattern: "^[4-5][0-9]{2}$"
```

Comment:
I don't see the value of including a pattern in a response object. Are we expecting the client to error check our error responses? Status should really be an integer anyway, but if it is to be a string, the client doesn't need to know any more than that.

Reply:
In the guidelines it appears as string but it could be an integer as well to me. Regarding patterns, we have patterns in other objects returned in responses. To me, it guides the implementation and enforces them to restrict values. 
All of this should be reflected in the guidelines so all APIs in the family design error responses in the same way.

Reply:
Two comments from my side: 

### 1. String or Integer

 @eric-murray: Based on the API design guideline for ErrorResponses, the status is expected as a String. 
``` json
{
   "status": "400",
   "code": "INVALID_ARGUMENT",
   "message": "A human readable description of what the event represent"
}
```
(see: https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md#6-error-responses)

otherwise it would be with integer a example like that:

``` json
{
   "status": 400,
   "code": "INVALID_ARGUMENT",
   "message": "A human readable description of what the event represent"
}
```

### 2. Pattern

@jlurien: I agree that a specific pattern-definition for Http-Status for 4xx or 5xx is not necessarry / I do not see the value, too.

Reply:
I agree that a common approach amongst all CAMARA APIs is preferable. But let's resolve the specific issues for this API, and then the template established here can be pushed to Commonalities.

On this issue, there is no right or wrong answer. I prefer a "minimalist" API specification that still allows the typical developer to use the API (*"Make everything as simple as possible, but not simpler"*, attributed to Einstein). But others may prefer to know that converting our status string to a short integer is perfectly reasonable, or that they don't have to worry about exceeding `maxsize` when storing the string. It's personal preference.

Reply:
> Based on the API design guideline for ErrorResponses, the status is expected as a String.

Sure, that is clear, but doesn't change my point that an integer type would be a better fit for what is, after all, known to be a 3 digit integer. And it could be changed, just not by this WG. Interpreting everything as a string, whilst usually successful, also does not usually help much with the next step of processing that data.

Reply:
RFC 9110 says clearly: [The status code of a response is a three-digit integer code](https://www.rfc-editor.org/rfc/rfc9110.html#name-status-codes:~:text=The%20status%20code%20of%20a%20response%20is%20a%20three%2Ddigit%20integer%20code). My proposal to open an issue at Commonalities for that (I can do that) and go in this PR with integer for the status code. In a sense we are "testing" the guidelines here.

Reply:
@hdamker: I also agree with Integer for this, but if we allign on that then @jlurien has to update all examples where "status" is marked currently as a string-value.

Reply:
Fine with integer, that ends the discussion for pattern as well, but better to move the discussion to commonalities, so other colleagues who don't follow this WG are aware of it. In the end I'll align the PR with all we agree.

Reply:
Opened https://github.com/camaraproject/WorkingGroups/issues/143 on the discussion String vs Integer. 

But: do we use the right field for the numeric HTTP status code? Shouldn't it go into "code" instead of "status"?

[OpenAPI Spec](https://spec.openapis.org/oas/v3.1.0#response-object): 
```
{
  "components": {
    "schemas": {
      "ErrorModel": {
        "type": "object",
        "required": [
          "message",
          "code"
        ],
        "properties": {
          "message": {
            "type": "string"
          },
          "code": {
            "type": "integer",
            "minimum": 100,
            "maximum": 600
          }
        }
      }, ...
```
So "code" is the HTTP status code as Integer. No "status" here ... but that can be an addition.

Shortly skimmed through the Camara sub projects ... no consistent picture.

Reply:
One is really the `status_code` while the other is thought as the `error_code`, the latest intended to enumerate some logical errors of the API, which could help clients to react better to them, as messages are human friendly but not convenient for programming logic.  For example MSISDN_REQUIRED, IP_ADDRESS_REQUIRED, QOS_PROFILE_NOT_SUPPORTED, etc. So far we are not defining a list of error codes, and in the end we only have one generic error code per status code, so is somehow redundant.

Reply:
It seems that we are going with the integer, so I'll update the PR accordingly.

Reply:
@jlurien: The issue in commonalities has been closed and the guideline changed accordingly (PR https://github.com/camaraproject/WorkingGroups/pull/146). So I suppose, we are good to finalize this PR here?

Reply:
@hdamker updated, please see my latest comment.

Reply:
@jlurien Thanks!